### PR TITLE
irmin: expose Tree.is_empty

### PR DESCRIPTION
**Pull request by CraigFe**

### CODE REVIEW

**Review by Ngoguey42 — APPROVED** [review]

**Comment by samoht** [comment]
LGTM but if we expose this we should make sure the result is consistent with the ones used with `repr` ; can we have tests for this?

**Comment by CraigFe** [comment]
I agree it's worth having tests for this, even if the edge cases are a bit tedious to construct. Will do that now & we can look into fuzz-testing this API at some later point.

**Comment by CraigFe** [comment]
I've changed the implementation of `is_empty` to pre-compute the hash of the empty tree, meaning it now
returns `bool` rather than `bool Lwt.t`, and also added some tests for equivalence with `Type.equal tree_t Tree.empty`.

File: src/irmin/tree.ml [hunk 1]
```diff
@@ -663,18 +663,17 @@ module Make (P : Private.S) = struct
...
-    let is_empty t =
-      match cached_map t with
-      | Some m -> Lwt.return (Ok (StepMap.is_empty m))
-      | None -> (
-          match t.v with
-          | Value (_, v, Some um) ->
-              let res = is_empty_after_updates v um in
-              Lwt.return_ok res
-          | _ -> (
-              to_value t >|= function
```
Comment (Ngoguey42) [inline src/irmin/tree.ml:675]: I'm not sure it matters but `to_value` has the advantage of testing `cached_value`

Comment (Ngoguey42) [inline src/irmin/tree.ml:675]: I'm not sure it matters but `to_value` has the advantage of testing `cached_value`

Comment (CraigFe) [reply src/irmin/tree.ml:675]: I'm not sure either, but for consistency with `cached_map` I've added a call to `cached_value`.

There must be a better way to handle this caching; will think about that in a later refactor.
